### PR TITLE
Rearranged stack analyses GET code for handling partial response gracefully

### PR DESCRIPTION
**Pull request by samuzzal-choudhury**

### CODE REVIEW

**Comment by samuzzal-choudhury** [comment]
Related issue: https://github.com/fabric8-analytics/fabric8-analytics-server/issues/18

File: bayesian/api_v1.py [hunk 1]
```diff
@@ -275,46 +275,41 @@ def get(ecosystem, package, version):
...
-                raise HTTPError(202, "Analysis for request ID '{t}' is in progress".format(t=external_request_id))
-        except SQLAlchemyError:
+        stack_result = retrieve_worker_result(rdb, external_request_id, "stack_aggregator")
+        reco_result = retrieve_worker_result(rdb, external_request_id, "recommendation")
+
+        if stack_result == None and reco_result == None:
+            raise HTTPError(202, "Analysis for request ID '{t}' is in progress".format(t=external_request_id))
+
+        if stack_result == -1 and reco_result == -1:
             raise HTTPError(500, "Worker result for request ID '{t}' doesn't exist yet".format(t=external_request_id))
```
Comment (msrb) [inline bayesian/api_v1.py]: This could be 404 instead of 500 (?)

File: bayesian/api_v1.py [hunk 2]
```diff
@@ -275,46 +275,41 @@ def get(ecosystem, package, version):
...
+        manifest_response = []
+        recommendation = {}
+
+        if stack_result != None and 'task_result' in stack_result:
+            if stack_result["task_result"] != None:
+                started_at = stack_result["task_result"]["_audit"]["started_at"]
+                finished_at = stack_result["task_result"]["_audit"]["ended_at"]
+                manifest_response.append(stack_result["task_result"])
+
+        if reco_result != None and 'task_result' in recommendation_result:
```
Comment (msrb) [inline bayesian/api_v1.py]: s/recommendation_result/reco_result

File: bayesian/api_v1.py [hunk 3]
```diff
@@ -275,46 +275,41 @@ def get(ecosystem, package, version):
...
-                                 .filter(WorkerResult.external_request_id == external_request_id,
-                                         or_(WorkerResult.worker == "stack_aggregator",
-                                             WorkerResult.worker == "recommendation"))
-            if results.count() <= 0:
-                raise HTTPError(202, "Analysis for request ID '{t}' is in progress".format(t=external_request_id))
-        except SQLAlchemyError:
+        stack_result = retrieve_worker_result(rdb, external_request_id, "stack_aggregator")
+        reco_result = retrieve_worker_result(rdb, external_request_id, "recommendation")
+
+        if stack_result == None and reco_result == None:
```
Comment (msrb) [inline bayesian/api_v1.py]: Not really a problem, but `if stack_result is None ...` would be more pythonic :)

File: bayesian/utils.py [hunk 4]
```diff
@@ -327,3 +327,19 @@ def fetch_public_key(app):
...
     return app.public_key
+
+def retrieve_worker_result (rdb, external_request_id, worker):
+    try:
+        results = rdb.session.query(WorkerResult)\
+                             .filter(WorkerResult.external_request_id == external_request_id,
+                             WorkerResult.worker == worker)
+        if results.count() <= 0:
+            return None
+    except SQLAlchemyError:
```
Comment (msrb) [inline bayesian/utils.py]: missing import
`from sqlalchemy.exc import SQLAlchemyError`

File: bayesian/utils.py [hunk 5]
```diff
@@ -327,3 +327,19 @@ def fetch_public_key(app):
             app.public_key = app.config.get('BAYESIAN_PUBLIC_KEY')
 
     return app.public_key
+
+def retrieve_worker_result (rdb, external_request_id, worker):
+    try:
+        results = rdb.session.query(WorkerResult)\
+                             .filter(WorkerResult.external_request_id == external_request_id,
+                             WorkerResult.worker == worker)
```
Comment (msrb) [inline bayesian/utils.py]: misleading indent, makes it hard(er) to read

**Comment by msrb** [comment]
LGTM, thanks 😉

**Pull request merged**